libsolc/libsolc.h

```diff
@@ -42,9 +43,11 @@ extern "C" {
 /// @param o_contents A pointer to the contents of the file, if found.
 /// @param o_error A pointer to an error message, if there is one.
 ///
-/// If the callback is not supported, o_contents and o_error should be set to NULL.
+/// The file (as well as error) contents that is to be allocated by the callback implementor must use the
+/// solidity_alloc() API to allocate its underlying storage and use solidity_free() for explicit release,
```

Comment:
This is wrong. Ownership of any allocation returned by the callback is transferred to the compiler.

Reply:
So deallocation is taken care of by the compiler then?

Reply:
Yes, just as currently (but the interface is more explicit about it).

Reply:
So, if the compiler takes ownership, in what situations will it NOT take ownership and you have to use solidity_reset()?

Reply:
Also, then this line is also wrong?
```
/// Use this function inside callbacks to allocate data that is to be passed to the compiler,
```
in line 65?

Reply:
Okay, @ekpyron explained it to me